src/renderers/gl/renderer.cpp

```diff
@@ -377,24 +377,35 @@ auto mrg::Renderer::render(mg::RenderableList const& renderables) const -> std::
 
 void mrg::Renderer::draw(mg::Renderable const& renderable) const
 {
+    auto const texture = gl_interface->as_texture(renderable.buffer());
     auto const clip_area = renderable.clip_area();
     if (clip_area)
     {
         glEnable(GL_SCISSOR_TEST);
-        glScissor(
-            clip_area.value().top_left.x.as_int() -
-                viewport.top_left.x.as_int(),
-            viewport.top_left.y.as_int() +
+
+        int y;
+        if (output_surface->layout() == graphics::gl::OutputSurface::Layout::TopRowFirst
+            && texture->layout() != mg::gl::Texture::Layout::TopRowFirst)
```

Comment:
Is this actually the correct test? I know it covers the case you've hit but wouldn't this be more general?
```suggestion
        if ((output_surface->layout() == graphics::gl::OutputSurface::Layout::TopRowFirst) !=
            (texture->layout() == mg::gl::Texture::Layout::TopRowFirst))
```